frontend/screens/FilterScreen.tsx

```diff
@@ -83,10 +83,10 @@ function FilterScreen({ navigation }) {
         </View>
         <Carousel
           title={"Genre"}
-          description={"Choose a mood from the ones below"}
+          description={"Choose a genre from the ones below"}
           choices={genreChoices}
-          selectedChoice={genre}
-          onChange={(choice) => { setGenre(choice) }}
+          selectedChoice={genres}
+          onChange={(choice) => { (choice < 0 || Object.is(choice, -0)) ? setGenres(genres & ~(1 << -choice)) : setGenres(genres | (1 << choice)) }}
```

Comment:
choice will be negative or -0 if it was already selected and needs to be unselected.